### PR TITLE
llvm: Disable assertions by default

### PR DESCRIPTION
**Pull request by aw1621107**
This makes the build consistent with std_cmake_args, which passes
-DNDEBUG. Before this both -DNDEBUG and -DLLVM_ENABLE_ASSERTIONS=On were
passed.

Forgot to change this when I added -DNDEBUG to std_cmake_args, I think.

Also not quite sure what to do with the deprecated options; do those need to be negated as well?

### CODE REVIEW

File: Library/Formula/llvm.rb [hunk 1]
```diff
@@ -116,7 +116,7 @@ class Llvm < Formula
   option "with-lldb", "Build LLDB debugger"
   option "with-python", "Build Python bindings against Homebrew Python"
   option "with-rtti", "Build with C++ RTTI"
-  option "without-assertions", "Speeds up LLVM, but provides less debug information"
+  option "with-assertions", "Provides more debug information, but slows down LLVM"
 
   deprecated_option "rtti" => "with-rtti"
   deprecated_option "disable-assertions" => "without-assertions"
```
Comment (DomT4) [inline Library/Formula/llvm.rb]: I think this line is safe to remove now. It won't be doing anything at least given the option is now off-by-default.

CC @mikemcquaid

Comment (aw1621107) [inline Library/Formula/llvm.rb]: I assume the deprecated_option for assertions should also be removed?

Comment (DomT4) [inline Library/Formula/llvm.rb]: I think it's fine, feel free.

**Comment by mikemcquaid** [comment]
Thanks for your contribution to Homebrew! Without people like you submitting PRs we couldn't run this project. You rock!

**Pull request closed without merging**